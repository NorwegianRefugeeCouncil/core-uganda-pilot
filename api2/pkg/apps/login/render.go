package login

import "net/http"

func (s *Server) Render(w http.ResponseWriter, req *http.Request, templateName string, data map[string]interface{}) {
	if err := s.template.ExecuteTemplate(w, templateName, data); err != nil {
		http.Error(w, err.Error(), http.StatusInternalServerError)
		return
	}
}
