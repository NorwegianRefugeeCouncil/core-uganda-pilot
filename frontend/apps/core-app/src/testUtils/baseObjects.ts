import { FormDefinition, FormType, Record } from 'core-api-client';

export const baseForm: FormDefinition = {
  id: 'form1',
  code: '',
  name: 'form 1',
  databaseId: 'databaseId1',
  folderId: 'folderId1',
  formType: FormType.DefaultFormType,
  fields: [],
};

export const baseTextField = {
  id: 'field1',
  name: 'field 1',
  description: 'description 1',
  code: '',
  required: false,
  key: false,
  fieldType: {
    text: {},
  },
};

export const baseMultilineTextField = {
  id: 'field2',
  name: 'field 2',
  description: 'description 2',
  code: '',
  required: true,
  key: true,
  fieldType: {
    multilineText: {},
  },
};

export const baseRecord1: Record = {
  id: 'record1',
  values: [
    {
      value: 'text',
      fieldId: 'field1',
    },
    {
      value: 'multi line text',
      fieldId: 'field2',
    },
  ],
  formId: 'form1',
  databaseId: 'databaseId1',
  ownerId: undefined,
};
