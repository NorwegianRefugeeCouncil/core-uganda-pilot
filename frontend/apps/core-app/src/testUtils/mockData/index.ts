export { makeRecord } from './Record';
export { makeForm } from './Form';
export { makeField } from './FormField';
