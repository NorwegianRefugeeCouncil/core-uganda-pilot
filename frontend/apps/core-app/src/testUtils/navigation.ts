export const mockNavigationProp = {
  dispatch: jest.fn(),
  navigate: jest.fn(),
  reset: jest.fn(),
  goBack: jest.fn(),
  isFocused: jest.fn(),
  canGoBack: jest.fn(),
  getParent: jest.fn(),
  getState: jest.fn(),
  setParams: jest.fn(),
  setOptions: jest.fn(),
  addListener: jest.fn(),
  removeListener: jest.fn(),
  replace: jest.fn(),
  push: jest.fn(),
  pop: jest.fn(),
  popToTop: jest.fn(),
};
