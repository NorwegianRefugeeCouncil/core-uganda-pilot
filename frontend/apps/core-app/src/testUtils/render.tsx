import * as React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import {
  render as r,
  RenderAPI,
  RenderOptions,
} from '@testing-library/react-native';
import { render as rWeb } from '@testing-library/react';
import { NativeBaseProvider } from 'native-base';
import { theme } from 'core-design-system';

const inset = {
  frame: { x: 0, y: 0, width: 0, height: 0 },
  insets: { top: 0, left: 0, right: 0, bottom: 0 },
};

export const render = (
  component: React.ReactElement<any>,
  options?: RenderOptions | undefined,
): RenderAPI =>
  r(
    <NativeBaseProvider theme={theme} initialWindowMetrics={inset}>
      <NavigationContainer>{component}</NavigationContainer>
    </NativeBaseProvider>,
    options,
  );

export const renderWeb = (
  component: React.ReactElement<any>,
  options?: RenderOptions | undefined,
) =>
  rWeb(
    <NativeBaseProvider theme={theme} initialWindowMetrics={inset}>
      <NavigationContainer>{component}</NavigationContainer>
    </NativeBaseProvider>,
    options,
  );
