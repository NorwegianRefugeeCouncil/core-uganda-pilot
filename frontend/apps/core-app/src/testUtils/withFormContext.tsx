import { Button } from 'native-base';
import React from 'react';
import { FieldValues, FormProvider, useForm } from 'react-hook-form';

export const withFormContext =
  <T,>(
    Component: React.FC<T>,
    defaultValues: FieldValues = {},
    onSubmit: (data: any) => void = () => {},
  ) =>
  // eslint-disable-next-line react/display-name
  (props: T) => {
    const f = useForm({ defaultValues });

    const handleSubmit = f.handleSubmit((data: any) => {
      onSubmit(data);
    });

    return (
      <FormProvider {...f}>
        <Component {...props} />
        <Button testID="with-form-context-submit-button" onPress={handleSubmit}>
          Submit
        </Button>
      </FormProvider>
    );
  };
