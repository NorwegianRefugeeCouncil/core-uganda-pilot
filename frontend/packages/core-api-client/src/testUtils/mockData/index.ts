export * from './Form';
export * from './FormField';
export * from './Record';
