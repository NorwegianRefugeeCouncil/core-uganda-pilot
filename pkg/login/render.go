package login

import "net/http"

func (s *Server) Render(w http.ResponseWriter, templateName string, data map[string]interface{}) {
	if err := s.template.ExecuteTemplate(w, templateName, data); err != nil {
		s.Error(w, err)
		return
	}
}
