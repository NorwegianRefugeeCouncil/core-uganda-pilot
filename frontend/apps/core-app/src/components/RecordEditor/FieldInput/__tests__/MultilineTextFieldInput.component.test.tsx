import { fireEvent, waitFor } from '@testing-library/react-native';

import { render } from '../../../../testUtils/render';
import { withFormContext } from '../../../../testUtils/withFormContext';
import { MultilineTextFieldInput } from '../MultilineTextFieldInput.component';

const makeDefaultValues = (value: string) => ({
  'form-id': {
    'field-id': value,
  },
});

const field = {
  id: 'field-id',
  name: 'Field name',
  description: 'Field description',
  code: '',
  required: false,
  key: false,
  fieldType: { multilineText: {} },
};

it('should match the snapshot', () => {
  const TextWithContext = withFormContext(
    MultilineTextFieldInput,
    makeDefaultValues('an input value'),
  );
  const { toJSON } = render(<TextWithContext formId="form-id" field={field} />);
  expect(toJSON()).toMatchSnapshot();
});

it('should handle on change', async () => {
  const onSubmitSpy = jest.fn();
  const TextWithContext = withFormContext(
    MultilineTextFieldInput,
    makeDefaultValues(''),
    onSubmitSpy,
  );
  const { getByTestId } = render(
    <TextWithContext formId="form-id" field={field} />,
  );
  const submitButton = getByTestId('with-form-context-submit-button');
  const text = getByTestId('multiline-text-field-input');

  fireEvent.press(submitButton);
  await waitFor(() => expect(onSubmitSpy).toHaveBeenCalledTimes(1));
  expect(onSubmitSpy).toHaveBeenCalledWith({
    'form-id': {
      'field-id': '',
    },
  });

  fireEvent.changeText(text, 'a new value');
  fireEvent.press(submitButton);
  await waitFor(() => expect(onSubmitSpy).toHaveBeenCalledTimes(2));
  expect(onSubmitSpy).toHaveBeenCalledWith({
    'form-id': {
      'field-id': 'a new value',
    },
  });
});
