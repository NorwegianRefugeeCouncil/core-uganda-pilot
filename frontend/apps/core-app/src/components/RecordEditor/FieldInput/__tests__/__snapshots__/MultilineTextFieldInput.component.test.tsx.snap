// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    dataSet={Object {}}
    isInvalid={false}
    isRequired={false}
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <View
      dataSet={Object {}}
      feedbackId="field-react-aria-1-feedback"
      hasFeedbackText={true}
      hasHelpText={true}
      helpTextId="field-react-aria-1-helptext"
      isDisabled={false}
      isInvalid={false}
      isReadOnly={false}
      isRequired={false}
      labelId="field-react-aria-1-label"
      nativeID="field-react-aria-1-label"
      setHasFeedbackText={[Function]}
      setHasHelpText={[Function]}
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "marginBottom": 2,
          "marginRight": 12,
        }
      }
    >
      <Text
        dataSet={Object {}}
        level="1"
        style={
          Object {
            "backgroundColor": undefined,
            "color": "#666666",
            "fontFamily": "Roboto_400Regular",
            "fontSize": 16,
            "fontStyle": undefined,
            "fontWeight": undefined,
            "letterSpacing": 0,
            "lineHeight": 23,
            "textAlign": "auto",
            "textDecorationLine": undefined,
          }
        }
      >
        Field name
      </Text>
    </View>
    <TextInput
      _text={
        Object {
          "color": "darkText",
        }
      }
      accessibilityDescribedBy="field-react-aria-1-feedback field-react-aria-1-helptext"
      accessible={true}
      allowFontScaling={true}
      dataSet={Object {}}
      disabled={false}
      editable={true}
      isRequired={false}
      multiline={true}
      nativeID="field-react-aria-1"
      numberOfLines={4}
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      onKeyPress={[Function]}
      placeholderTextColor="#a3a3a3"
      readOnly={false}
      rejectResponderTermination={true}
      required={false}
      secureTextEntry={false}
      selectionColor="#1f2937"
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
            "borderColor": "#666666",
            "borderRadius": 3,
            "borderStyle": "solid",
            "borderWidth": 1,
            "color": "#666666",
            "fontFamily": "Roboto_400Regular",
            "fontSize": 18,
            "fontStyle": undefined,
            "fontWeight": undefined,
            "height": 80,
            "lineHeight": 26,
            "overflow": "hidden",
            "paddingBottom": 8,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 8,
            "width": undefined,
          },
          Object {},
        ]
      }
      testID="multiline-text-field-input"
      textAlignVertical="top"
      underlineColorAndroid="transparent"
      value="an input value"
    />
    <View
      dataSet={Object {}}
      feedbackId="field-react-aria-1-feedback"
      hasFeedbackText={true}
      hasHelpText={true}
      helpTextId="field-react-aria-1-helptext"
      isDisabled={false}
      isInvalid={false}
      isReadOnly={false}
      isRequired={false}
      labelId="field-react-aria-1-label"
      nativeID="field-react-aria-1-feedback"
      setHasFeedbackText={[Function]}
      setHasHelpText={[Function]}
      style={
        Object {
          "marginTop": 2,
        }
      }
    >
      <Text
        dataSet={Object {}}
        level="1"
        style={
          Object {
            "backgroundColor": undefined,
            "color": "#666666",
            "fontFamily": "Roboto_400Regular",
            "fontSize": 14,
            "fontStyle": undefined,
            "fontWeight": undefined,
            "letterSpacing": 0,
            "lineHeight": 20,
            "textAlign": "auto",
            "textDecorationLine": undefined,
          }
        }
      >
        Field description
      </Text>
    </View>
  </View>
  <View
    _text={
      Object {
        "color": "lightText",
      }
    }
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    colorScheme="primary"
    dataSet={Object {}}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onHoverIn={[Function]}
    onHoverOut={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FF7602",
        "borderRadius": 3,
        "flexDirection": "row",
        "height": 50,
        "justifyContent": "center",
        "paddingBottom": 8,
        "paddingLeft": 50,
        "paddingRight": 50,
        "paddingTop": 8,
      }
    }
    testID="with-form-context-submit-button"
  >
    <View
      dataSet={Object {}}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        dataSet={Object {}}
        style={Object {}}
      >
        <Text
          dataSet={Object {}}
          level="1"
          style={
            Object {
              "backgroundColor": undefined,
              "color": "#FFFFFF",
              "fontFamily": "Roboto_500Medium",
              "fontSize": 20,
              "fontStyle": undefined,
              "fontWeight": undefined,
              "letterSpacing": 0,
              "lineHeight": 24,
              "textAlign": "auto",
              "textDecorationLine": undefined,
            }
          }
        >
          Submit
        </Text>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
